plugins/header_rewrite/header_rewrite.cc

```diff
@@ -222,16 +223,21 @@ RulesConfig::parse_config(const std::string &fname, TSHttpHookID default_hook)
       }
     }
 
-    if (p.is_cond()) {
-      if (!rule->add_condition(p, filename.c_str(), lineno)) {
-        delete rule;
-        return false;
-      }
-    } else {
-      if (!rule->add_operator(p, filename.c_str(), lineno)) {
-        delete rule;
-        return false;
+    // This is pretty ugly, but it turns out, some conditions / operators can fail (regexes), which didn't use to be the case.
+    // Long term, maybe we need to perculate all this up through add_condition() / add_operator() rather than this big ugly try.
+    try {
```

Comment:
Try and catch?  What the hell are you doing?  :)


j/k - I am OK with it considering the alternative.